CoreScriptsRoot/Modules/SocialUtil.lua

```diff
@@ -0,0 +1,124 @@
+--[[
+	// FileName:    SocialUtil.lua
+	// Written by:  TheGamer101
+	// Description: Utility code related to social features.
+]]
+local SocialUtil = {}
+
+--[[ Services ]]--
+local Players = game:GetService("Players")
+
+--[[ Constants ]]--
+local THUMBNAIL_SIZE_MAP = {
+	[Enum.ThumbnailSize.Size48x48]   =  48,
+	[Enum.ThumbnailSize.Size180x180] = 180,
+	[Enum.ThumbnailSize.Size420x420] = 420
+}
+
+local THUMBNAIL_FALLBACK_URLS = {
+	[Enum.ThumbnailType.HeadShot] = "https://www.roblox.com/headshot-thumbnail/image?width=%d&height=%d&format=png&userId=%d",
+	[Enum.ThumbnailType.AvatarBust] = "https://www.roblox.com/bust-thumbnail/image?width=%d&height=%d&format=png&userId=%d",
+	[Enum.ThumbnailType.AvatarThumbnail] = "https://www.roblox.com/avatar-thumbnail/image?width=%d&height=%d&format=png&userId=%d"
+}
+
+local GET_PLAYER_IMAGE_DEFAULT_TIMEOUT = 5
+local DEFAULT_THUMBNAIL_SIZE = Enum.ThumbnailSize.Size48x48
+local DEFAULT_THUMBNAIL_TYPE = Enum.ThumbnailType.AvatarThumbnail
+local GET_USER_THUMBNAIL_ASYNC_RETRY_TIME = 1
+
+--[[ Utility ]]--
+local function CreateSignal()
```

Comment:
I could use the the definition from the Utility module in Settings but that seemed weird to me, that Module is very Settings specific and doesn't look like it should have external dependencies.